pkg/odo/cli/component/common_link.go

```diff
@@ -40,6 +39,8 @@ type commonLinkOptions struct {
 	secretName       string
 	isTargetAService bool
 
+	devfilePath string
```

Comment:
Rather than adding `devfilePath` to each struct for the commands that are missing it or need it, could we add it to something common like `genericclioptions.Context`?

Doing so might require slightly retooling the initilization of the contexts, however.

Reply:
Yeah, we can make this change in a later PR as it'd thorooughly mess up my constant rebasing :( I do know that Chris has done refactoring he'd like to make in terms of Devfile / context usage.

Reply:
Ah good point, we can handle that either as part of the context refactoring or later on 👍